semeio/jobs/rft/gendata_rft.py

```diff
@@ -62,8 +63,15 @@ def _populate_trajectory_points(
     return trajectory_points
 
 
-def run(well_times, trajectories, ecl_grid, ecl_rft, zonemap=None):
-
+def run(
+    well_times,
+    trajectories,
+    ecl_grid,
+    ecl_rft,
+    zonemap=None,
+    csvfile="_gendata_rft.csv",
```

Comment:
Shouldn´t this be the other way around? By that I mean, you are now setting it ON as default, shouldn´t it be opt-in instead of opt-out?, this should be `None` as default?

Reply:
I think I want this ON by default, enabling a transition to a situation where the dumped CSV file is the authoritative source for these simulation results (and not the scattered text files).

Reply:
All defaults should be handled in the `argparse` section. In that case the user can be informed of defaults by running `app_name --help`. If we add the argparse to documentation it will be shown there as well. I suppose if we change the default in `argparse` it will always override the one added here as well, making it irrelevant to the code.

Reply:
Yes, I will remove it from  `jobs/rft/gendata_rft.py` and inject it as a default parameter in `jobs/config_jobs/GENDATA_RFT`

Reply:
Ah, yes, I hate that part.. Having to add it to the `JOB_DESCRIPTION` file, where none of that information is propagated to the user.. I really disliked it when we did it with `ZONEMAP` as well. The `JOB_DESCRIPTION` file doesn't handle optional parameters very well, does it.

Wouldn't need it in the `argparse` either then.

Reply:
Here is how the optional parameter is handled for `ZoneMap` https://github.com/equinor/semeio/blob/138d1b666c4efbe033a9dc63c739736247d3560e/semeio/jobs/rft/zonemap.py#L21
I believe that is how it was handled before we did the refactoring as well.

Reply:
Mitigate it by syncing up JOB_DESCRIPTION and argparse behaviour is perhaps the best we can do for now. Noone (i.e.  ert 2.x users)  is using the argparse interface directly anyway except developers.

Reply:
Changing the `run()` default to `None`, letting it be overridden by both argparse default and `JOB_DESCRIPTION`.

Reply:
> Noone (i.e. ert 2.x users) is using the argparse interface directly anyway except developers.

But we will likely build the documentation on top of the `argparse`, and thus it will be shown indirectly to the user (this has been done for several of the jobs in everest, and will likely also be done for ert jobs). So it is good that the `argparse` instance is well-made.

Reply:
> Mitigate it by syncing up JOB_DESCRIPTION and argparse behaviour is perhaps the best we can do for now. Noone (i.e. ert 2.x users) is using the argparse interface directly anyway except developers.

Then it would be good to add a test to keep the two in sync.

Reply:
How? There is not test infrastructure in the repository to test ert on ert config files (?)

Reply:
> How? There is not test infrastructure in the repository to test ert on ert config files (?)

Not really, we might consider adding that, I was thinking just a quick and dirty comparison by pulling it from argparse and comparing with the `JOB_DESCRIPTION`.

Reply:
Done.